File: Java/Merge_Intervals.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Comparator;
import java.util.Iterator;
import java.util.List;

public class MergeIntervals {
    public static List<List<Integer>> merge(List<List<Integer>> intervals) {
        intervals.sort(Comparator.comparingInt(x -> x.get(0)));

        
        Iterator<List<Integer>> iterator = intervals.iterator();
        int i = 0;
        while (iterator.hasNext()) {
            List<Integer> current = iterator.next();

            if (iterator.hasNext()) {
                List<Integer> next = iterator.next();

                if (current.get(1) >= next.get(0)) {
                    current.set(1, Math.max(current.get(1), next.get(1)));
                    iterator.remove();
                } else {
                    i++;
                }
            }
        }

        return intervals;
    }

    public static void main(String[] args) {
        
        List<List<Integer>> intervals = new ArrayList<>(Arrays.asList(
                Arrays.asList(1, 3),
                Arrays.asList(2, 6),
                Arrays.asList(8, 10),
                Arrays.asList(15, 18)
        ));

        List<List<Integer>> result = merge(intervals);
        System.out.println(result);
    }
}

File: Python/Merge Intervals.py
def merge(intervals):
    intervals.sort(key=lambda x: x[0])

    i = 0
    while i < len(intervals) - 1:
        if intervals[i][1] >= intervals[i+1][0]:
            intervals[i] = [intervals[i][0], max(
                intervals[i][1], intervals[i+1][1])]
            intervals.pop(i + 1)
        else:
            i += 1

    return intervals


intervals = [[1, 3], [2, 6], [8, 10], [15, 18]]
print(merge(intervals))
